app/components/form/SideModalForm.tsx

```diff
@@ -15,8 +15,8 @@ import { Button } from '~/ui/lib/Button'
 import { SideModal } from '~/ui/lib/SideModal'
 
 type SideModalFormProps<TFieldValues extends FieldValues> = {
-  id: string
   form: UseFormReturn<TFieldValues>
+  formType: 'create' | 'edit'
```

Comment:
Check this out — you can make TS error if an edit form is given a `submitLabel`. We can always change it later if we decide we want to be able to override it. And look! It caught one!

```diff
diff --git a/app/components/form/SideModalForm.tsx b/app/components/form/SideModalForm.tsx
index 3ad1ed84..28c34d4c 100644
--- a/app/components/form/SideModalForm.tsx
+++ b/app/components/form/SideModalForm.tsx
@@ -16,7 +16,6 @@ import { SideModal } from '~/ui/lib/SideModal'
 
 type SideModalFormProps<TFieldValues extends FieldValues> = {
   form: UseFormReturn<TFieldValues>
-  formType: 'create' | 'edit'
   /**
    * A function that returns the fields.
    *
@@ -35,8 +34,17 @@ type SideModalFormProps<TFieldValues extends FieldValues> = {
   title: string
   subtitle?: ReactNode
   onSubmit?: (values: TFieldValues) => void
-  submitLabel?: string
-}
+} & (
+  | {
+      formType: 'edit'
+      // submitLabel is not allowed on edit forms. we want them all to say "save changes"
+      submitLabel?: never
+    }
+  | {
+      formType: 'create'
+      submitLabel?: string
+    }
+)
 
 /**
  * Only animate the modal in when we're navigating by a client-side click.
diff --git a/app/forms/subnet-edit.tsx b/app/forms/subnet-edit.tsx
index 98137a6c..3fe0d924 100644
--- a/app/forms/subnet-edit.tsx
+++ b/app/forms/subnet-edit.tsx
@@ -48,7 +48,6 @@ export function EditSubnetForm({ onDismiss, editing }: EditSubnetFormProps) {
       }}
       loading={updateSubnet.isPending}
       submitError={updateSubnet.error}
-      submitLabel="Update subnet"
     >
       <NameField name="name" control={form.control} />
       <DescriptionField name="description" control={form.control} />
```

Reply:
nice